### PR TITLE
No git_url or git_tag

### PR DESCRIPTION
**Pull request by scopatz**
Checklist
* [X] Added a ``news`` entry

### CODE REVIEW

**Comment by scopatz** [comment]
Yeah, will do.

**Comment by scopatz** [comment]
done.

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -19,10 +19,7 @@
 import github
 import ruamel.yaml
 
-from conda_build.metadata import (
-    ensure_valid_license_family,
-    FIELDS as cbfields,
-)
+from conda_build.metadata import ensure_valid_license_family, FIELDS as cbfields
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]: Can we stop going back and forth on this? This was changed in https://github.com/conda-forge/conda-smithy/pull/1003 and now it's reverted here

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -91,13 +91,9 @@ def get_list_section(parent, name, lints, allow_single=False):
 
 
 def lint_section_order(major_sections, lints):
-    section_order_sorted = sorted(
-        major_sections, key=EXPECTED_SECTION_ORDER.index
-    )
+    section_order_sorted = sorted(major_sections, key=EXPECTED_SECTION_ORDER.index)
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]: This was changed by black and now it's reverted? I don't understand

File: conda_smithy/lint_recipe.py [hunk 3]
```diff
@@ -91,13 +91,9 @@ def get_list_section(parent, name, lints, allow_single=False):
...
     if major_sections != section_order_sorted:
-        section_order_sorted_str = map(
-            lambda s: "'%s'" % s, section_order_sorted
-        )
+        section_order_sorted_str = map(lambda s: "'%s'" % s, section_order_sorted)
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]: Same here

File: conda_smithy/lint_recipe.py [hunk 4]
```diff
@@ -111,8 +107,7 @@ def lint_about_contents(about_section, lints):
         # if the section doesn't exist, or is just empty, lint it.
         if not about_section.get(about_item, ""):
             lints.append(
-                "The {} item is expected in the about section."
-                "".format(about_item)
+                "The {} item is expected in the about section." "".format(about_item)
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]: Same here

File: conda_smithy/lint_recipe.py [hunk 5]
```diff
@@ -136,14 +131,12 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
-    
+
     # 0: Top level keys should be expected
     unexpected_sections = []
     for section in major_sections:
         if section not in EXPECTED_SECTION_ORDER:
-            lints.append(
-                "The top level meta key {} is unexpected".format(section)
-            )
+            lints.append("The top level meta key {} is unexpected".format(section))
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py]: Same here

**Comment by isuruf** [comment]
Maybe the change is because of different max number of characters per line limit used by @mariusvniekerk and you

File: conda_smithy/lint_recipe.py (hunk 2 above)
Comment (scopatz) [reply conda_smithy/lint_recipe.py]: I don't understand either maybe @mariusvniekerk  has his black setup to use longer / shorter line lengths?

File: conda_smithy/lint_recipe.py [hunk 6]
```diff
@@ -402,6 +385,14 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
             "[here](https://conda-forge.org/docs/maintainer/knowledge_base.html#compilers)."
         )
 
+    # 22:
+    for source_section in sources_section:
+        if "git_url" in source_section or "git_tag" in source_section:
+            lints.append(
+                "Building from git repositories is not allowed; please build from "
+                "a source tarball. [See the docs here.](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)"
+            )
```
Comment (jakirkham) [inline conda_smithy/lint_recipe.py]: Actually I'm not sure this is a good idea. AFAIK go packages in particular need to clone from GitHub.

cc @sodre

Comment (scopatz) [reply conda_smithy/lint_recipe.py]: Ahh that is a fair point, if we can confirm.

Comment (sodre) [reply conda_smithy/lint_recipe.py]: I can confirm that go packages are often built directly from git.

**Comment by scopatz** [comment]
Closing this as it does not apply to go packages :man_facepalming:

**Pull request closed without merging**